src/rascal/math/hyp1f1.hh

```diff
@@ -30,487 +30,185 @@
-                << " z=" << std::to_string(z) << std::endl;
-          throw std::overflow_error(error.str());
-        }
-        return res;
-      }
-    };
-
-    /**
-     * Computes the 1F1 with the asymptotic limit
-     * @f[
-     *      1F1(a,b,z) \sim \exp{z} z^{a-b} \frac{\Gamma{b}}{\Gamma{a}}
-     *                    \sum_{j=0}^{\infty} \frac{(b-a)_j(1-a)_j}{j!} z^{-j}
-     * @f]
-     *
-     * @f[
-     *  G(a,b,z) = \frac{\Gamma(a)}{\Gamma(b)} * \exp{-\alpha r_{ij}^2}
-     *                    * 1F1(a,b,z)
-     * @f]
-     */
-    class Hyp1f1Asymptotic {
-     private:
-      double a, b, prefac;
-      bool is_n_and_l{false};
-      bool is_exp{false};
-      double tolerance;
-      size_t mmax;
-      Eigen::VectorXd coeff{};
-      Eigen::VectorXd coeff_derivative{};
-
-      double z_power_a_b(double z) {
-        using math::pow;
-        double fac{0.};
-        if (this->is_n_and_l) {
-          fac = std::sqrt(pow(z, static_cast<int>(2 * (this->a - this->b))));
-        } else {
-          fac = pow(z, this->a - this->b);
-        }
-        return fac;
-      }
-
-     public:
-      // Parameter used to fix the sum when evaluating the numerical
-      // derivatives.
-      int n_terms{20};
-
-      Hyp1f1Asymptotic(double a, double b, size_t mmax,
-                       double tolerance = 1e-14)
-          : a{a}, b{b}, prefac{std::tgamma(b) / std::tgamma(a)},
-            tolerance{tolerance}, mmax{mmax} {
-        double intpart;
-        double f2{std::modf(2 * (a - b), &intpart)};
-        if (std::abs(f2) < 1e-14) {
-          this->is_n_and_l = true;
-        }
-
-        if (std::abs(1 - this->b / this->a) < DBL_FTOL) {
-          this->is_exp = true;
-        } else {
-          coeff.resize(mmax);
-          coeff_derivative.resize(mmax);
-          // precomputes the (1-a)_i / (b-a)_i / i! coefficients for the
-          // computation of hyp2f0
-          double bma{b - a}, oma{1 - a}, u{0.}, oma1{1 - a - 1};
-          coeff(0) = bma * oma;
-          coeff_derivative(0) = bma * oma1;
-          for (size_t i{1}; i < mmax; ++i) {
-            u = (bma + i) * (oma + i) / (i + 1);
-            coeff(i) = coeff(i - 1) * u;
-            u = (bma + i) * (oma1 + i) / (i + 1);
-            coeff_derivative(i) = coeff_derivative(i - 1) * u;
-          }
-        }
-      }
+      class Hyp1f1Series {
+       private:
+        double a, b;
+        size_t mmax;
+        double prefac;
+        double tolerance;
+        bool is_exp{false};
+
+        Eigen::VectorXd coeff{};
+        Eigen::VectorXd coeff_derivative{};
+
+       public:
+        // Parameter used to fix the sum when evaluating the numerical
+        // derivatives.
+        int n_terms{100};
+
+        Hyp1f1Series(double a, double b, size_t mmax, double tolerance = 1e-14);
+
+        /**
+         * Computes G(a,b,z)
+         * @param z -> pow(alpha * r_ij, 2) / (alpha + beta)
```

Comment:
Might as well wrap these math statements in `@f`s, just to be consistent, in case this ever gets rendered.

Reply:
Done!